### ISSUE TITLE
Transparent Pixels get their RGB values turned to white

### ISSUE THREAD

**Issue by armadillu** [fnordware/SuperPNG#6]
Hi, first of all, thanks for sharing the source of this! I am trying to find a solution modifying RGB values for transparent pixels (see attached image), and I came across this.

I managed to recompile the plugin in Xcode using your source, after a few changes to the photoshopsdk (some headers not found and such). I am unfamiliar with the PS plugin SDK, but I have a question.

I am wondering, when the plugin gets the pixel data hands off from Photoshop, is the data already "dirty" with the fully transparent pixiels turned to white? Or it's something that the plugin does? If so, is there a way to tell Photoshop not to do before you get the pixel data? I'm really looking for a way to disable this behavior.

Thanks!

![png-woes](https://cloud.githubusercontent.com/assets/167057/7403897/24e149e6-eea9-11e4-85ba-11f25ebc848d.jpg)

**Comment by fnordware** [comment]
Yeah, the RGB values for transparent pixels in Photoshop are sort of undefined. You never know what you might get. The "Clean Transparent" feature will turn them all black.

You could also put your Alpha in the channels palette and get full control over the RGB.
